### PR TITLE
Data collector

### PR DESCRIPTION
**Pull request by rjewing**
Added smooth fake data generation which is sent through unix domain sockets. The data collector receives from these sockets and updates the current values. The scheduler will then receive from this data collector and forward the values to the scheduled tasks.

The values are in the order of: patiend_id, bps, bpd, pulse, oxygen

TODO: Log alerts to database when David's database pull request (#2) is merged

### CODE REVIEW

File: alarm/alarm.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+from gmail_handler import send_email
+
+EMAIL_TO_NOTIFY = "rjewing@bu.edu"  # Set to the email you want to notify
```
Comment (YahiaBakour) [inline alarm/alarm.py]: Should we be pulling this from a db?

Comment (YahiaBakour) [reply alarm/alarm.py]: or maybe another file (what if you want to notify multiple people idk)?

**Review by YahiaBakour — APPROVED** [review]

**Comment by YahiaBakour** [comment]
Yo write up some docs into the README on all this pls

File: alarm/alarm.py (hunk 1 above)
Comment (rjewing) [reply alarm/alarm.py]: Yeah you're right, we should allow multiple emails and pull from a config/database.

File: alarm/alarm.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        self.threshold_low = threshold_low
+        self.threshold_high = threshold_high
+        self.name = name
+
+    def check_threshold(self, value):
+        if value is None:
+            return
+        if not self.threshold_low < value < self.threshold_high:
+            print(
+                f"Alarm {self.name} went out of range ({self.threshold_low}, {self.threshold_high}) with value {value}!")
```
Comment (dlhi) [inline alarm/alarm.py:16]: Can you provide a patient ID as well; a add to alert database call has to be added here

File: data_collector/data_vendor.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        t = Thread(target=self.receiver, daemon=True)
+        t.start()
+
+    def receiver(self):
+        while True:
+            self.bps = self.bp_sock.get_value()
+            self.bpd = self.bps - 40  # Fake a normal offset for diastolic
+            self.pulse = self.pulse_sock.get_value()
+            self.oxygen = self.oxygen_sock.get_value()
+            sleep(0.1)
```
Comment (dlhi) [inline data_collector/data_vendor.py]: Should increase the sleep time to account for lag between threads

**Review by chsclarke — APPROVED** [review]
refer to previous comments. looks great otherwise

File: data_collector/data_vendor.py (hunk 3 above)
Comment (YahiaBakour) [reply data_collector/data_vendor.py]: make it a config variable @rjewing

File: alarm/alarm.py (hunk 2 above)
Comment (rjewing) [reply alarm/alarm.py:16]: database call will be added once #2 is merged

**Review by YahiaBakour — APPROVED** [review]
Write up a doc and add a docs folder.

lgtm

**Pull request merged**